src/bsp/platform/stm32fx_nucleo_wifi/xi_bsp_io_net_stm32fx_nucleo_wifi.c

```diff
@@ -138,7 +149,7 @@ xi_bsp_io_net_state_t xi_bsp_io_net_read( xi_bsp_socket_t xi_socket,
     {
         return XI_BSP_IO_NET_STATE_ERROR;
     }
-    else if ( xi_net_state.wifi_state == xi_wifi_state_connected )
+    else if ( xi_net_state.wifi_state == xi_wifi_state_disconnected )
```

Comment:
No worries I missed that as well during the rewiew - I guess having unit tests or integration tests or functional tests would help us a lot during porting - it would speed up developing a production quality code.